subsys/net/lib/nrf_cloud/src/nrf_cloud_cell_pos.c

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <net/socket.h>
+#include <nrf_socket.h>
+
+#include <cJSON.h>
+#include <cJSON_os.h>
+#include <modem/modem_info.h>
+#include <net/nrf_cloud_cell_pos.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(nrf_cloud_cell_pos, CONFIG_NRF_CLOUD_LOG_LEVEL);
+
+#include "nrf_cloud_transport.h"
+#include "nrf_cloud_agps_schema_v1.h"
+
+#define CELL_POS_JSON_MSG_TYPE_KEY		"messageType"
+#define CELL_POS_JSON_MSG_TYPE_VAL_DATA		"DATA"
+
+#define CELL_POS_JSON_DATA_KEY			"data"
+#define CELL_POS_JSON_MCC_KEY			"mcc"
+#define CELL_POS_JSON_MNC_KEY			"mnc"
+#define CELL_POS_JSON_AREA_CODE_KEY		"tac"
+#define CELL_POS_JSON_CELL_ID_KEY		"eci"
+#define CELL_POS_JSON_PHYCID_KEY		"phycid"
+#define CELL_POS_JSON_TYPES_KEY			"types"
+#define CELL_POS_JSON_CELL_LOC_KEY_DOREPLY	"doReply"
+
+#define CELL_POS_JSON_APPID_KEY			"appId"
+#define CELL_POS_JSON_APPID_VAL_SINGLE_CELL	"SCELL"
+#define CELL_POS_JSON_APPID_VAL_MULTI_CELL	"MCELL"
+#define CELL_POS_JSON_CELL_LOC_KEY_LAT		"lat"
+#define CELL_POS_JSON_CELL_LOC_KEY_LON		"lon"
+#define CELL_POS_JSON_CELL_LOC_KEY_UNCERT	"uncertainty"
+
+static bool json_initialized;
+
+struct cell_based_loc_data {
```

Comment:
to be consistent, with file naming convention, suggest renaming to `nrf_cloud_cell_pos_result` or similar.